src/table/boot.rs

```diff
@@ -278,7 +278,17 @@ impl BootServices {
     /// The image must either successfully load an OS and call `ExitBootServices`
     /// in that time, or disable the watchdog.
     ///
-    /// Otherwise, the system will be reset after the time expires.
+    /// Otherwise, the firmware will log the event using the provided code and
+    /// data, then reset the system.
+    ///
+    /// The watchdog codes from 0 to 0xffff (65535) are reserved for internal
+    /// firmware use. You should therefore only use them if instructed to do so
+    /// by firmware-specific documentation. Higher values can be used freely.
+    ///
+    /// If provided, the watchdog data must be a null-terminated string
+    /// optionally followed by other binary data.
+    ///
+    /// FIXME: Validate the watchdog data
```

Comment:
@GabrielMajeri Do you already have some UTF-16 validation code handy for this?

Reply:
Not exactly. But we don't check in other places either, and the spec basically recommends applications to try and output text even if it contains some unsupported characters, since hardware *should* ignore them.

Also, UEFI uses the fixed-length UCS-2, not UTF-16 -- I'm not sure there's anything to validate. All `u16`s should map to some Unicode character.

Reply:
Then I guess the only validation we need to perform is to make sure that there is a zero in the data (as otherwise one might involuntarily send the firmware on an unbounded readout loop).

Reply:
Done in the latest commit